app/scripts/controllers/permissions/permission-log.test.js

```diff
@@ -667,14 +578,90 @@ describe('permissions log', function () {
         logMiddleware({ ...x.req }, { ...x.res });
       });
 
-      // validate history
-      permHistory = permLog.getHistory();
+      expect(permLog.getHistory()).toStrictEqual(EXPECTED_HISTORIES.case3[1]);
+    });
+  });
 
-      assert.deepEqual(
-        permHistory,
-        EXPECTED_HISTORIES.case3[1],
-        'should have expected history',
-      );
+  describe('updateAccountsHistory', () => {
+    beforeEach(() => {
+      initClock();
+    });
+
+    afterEach(() => {
+      tearDownClock();
+    });
+
+    it('does nothing if the list of accounts is empty', () => {
+      const permLog = initPermLog();
+      permLog.updateAccountsHistory('foo.com', []);
+
+      expect(permLog.getHistory()).toStrictEqual({});
+    });
+
+    it('updates the account history', () => {
+      const permLog = initPermLog({
+        permissionHistory: {
+          'foo.com': {
+            [PERM_NAMES.eth_accounts]: {
+              accounts: {
+                '0x1': 1,
+              },
+              lastApproved: 1,
+            },
+          },
+        },
+      });
+
+      clock.tick(1);
```

Comment:
What is this for? I didn't see any timer related code that would need this.

Reply:
It determines the number value of the account properties in the expected state object below.

Reply:
Ahh I see